docker/runtime/Dockerfile

```diff
@@ -38,51 +41,58 @@ RUN mkdir /CodeCompass-build && \
 #--------------------------    PRODUCTION STAGE   ----------------------------#
 ###############################################################################
 
-FROM ubuntu:20.04
+FROM ubuntu:22.04
 
 # tzdata package is installed implicitly in the following command. This package
 # sets timezone interactively during the installation process. This environment
 # variable prevents this interaction.
 ARG DEBIAN_FRONTEND=noninteractive
+ARG INSTALL_COMMAND
 
 ARG CC_DATABASE
 ENV CC_DATABASE ${CC_DATABASE}
 
-RUN if [ "pgsql" = "${CC_DATABASE}" ]; then \
-    apt-get update -qq --yes && \
-    apt-get install -qq --yes --no-install-recommends \
-    postgresql-server-dev-12 \   
-    libodb-pgsql-dev; \
-    else \
-    apt-get update -qq && \
-    apt-get install -qq --yes --no-install-recommends \
-    libsqlite3-dev \
-    libodb-sqlite-dev; \
-    fi;
+COPY --from=builder /packages/lib* /packages/
+
+RUN apt-get update -qq \
+  && ${INSTALL_COMMAND} \
+    /packages/libodb-2.5.deb \
+    /packages/libodb-dev.deb \
+  && if [ "pgsql" = "${CC_DATABASE}" ]; then \
+    ${INSTALL_COMMAND} \
+      libpq5 \
+      /packages/libodb-pgsql-2.5.deb \
+      /packages/libodb-pgsql-dev.deb \
```

Comment:
So you compiled these DEB packages on your computer and simply added them to the packages directory?

Reply:
@mozesl I gave a second thought to this after our previous meeting, and made a little research on the topic. I found that adding DEB files to Docker images like this is not the usual approach, instead a more favorable solution would be to build ODB and Thrift as part of the Docker image (as a layer).

Of course this could effect, efficiency, but there are caching tools in Docker to tackle this problem:
- When rebuilding the images locally, the layers from the existing previous build can be reused by Docker. Only the first layer, which changes and the ones after it have t be rebuild. See the [Optimizing builds with cache management guide](https://docs.docker.com/build/cache/) for more details if you are unfamiliar with it.
- Of course in CI the individual builds happen independent of each other and the build cache is not preserved between builds. However, with the usage of either the *inline*, the *registry* or the *gha* cache, we could easily reach the same result. The only difference would be that Docker image would not get a full clean build in the CI, with its benefits and drawbacks. (We could still enforce a clean build e.g. on releases.) See the [Cache storage backends guide](https://docs.docker.com/build/cache/backends/) or [this blog post](https://lipanski.com/posts/speed-up-your-docker-builds-with-cache-from) for further details.